include/configs/rockchip-common.h

```diff
@@ -62,13 +62,23 @@
 	BOOT_TARGET_DEVICES_references_MTD_without_CONFIG_CMD_MTD_BLK
 #endif
 
-/* First try to boot from SD (index 1), then NVME (if CMD_NVME is enabled), then eMMC (index 0) */
+/* First try to boot from SD (index 1), then NVME (if CMD_NVME is enabled), then SCSI (if CMD_SCSI is enabled),then eMMC (index 0) */
 #if CONFIG_IS_ENABLED(CMD_MMC)
```

Comment:
```suggestion
	#define BOOT_TARGET_PERSISTENT_STORAGE(func) \
#if CONFIG_IS_ENABLED(CMD_MMC)
		func(MMC, mmc, 1) \
#endif
#if CONFIG_IS_ENABLED(CMD_NVME)
		func(NVME, nvme, 0) \
		func(NVME, nvme, 1) \
#endif
#if CONFIG_IS_ENABLED(CMD_SCSI)
		func(SCSI, scsi, 0) \
#endif
#if CONFIG_IS_ENABLED(CMD_MMC)
		func(MMC, mmc, 0) \
#endif
		// empty clause in case nothing is enabled
```
更新一下宏的名字来更好的描述它的功能，同时需要修改后面的BOOT_TARGET_DEVICES(func)来调用这个新名字

Reply:
好像是不能再宏定义里面使用#if
include/configs/rockchip-common.h:66:44: error: '#' is not followed by a macro parameter
   66 | #define BOOT_TARGET_PERSISTENT_STORAGE(func) \
      |                                            ^
make[1]: *** [scripts/Makefile.autoconf:79: u-boot.cfg] Error 1
make: *** No rule to make target 'include/config/auto.conf', needed by 'include/config/uboot.release'.  Stop.

Reply:
那还是把BOOT_TARGET_MMC拆分了，然后在后面的`BOOT_TARGET_DEVICES`里面来指定启动顺序：
```c

#if CONFIG_IS_ENABLED(CMD_MMC)
	#define BOOT_TARGET_EMMC(func)		func(MMC, mmc, 0)
	#define BOOT_TARGET_MICROSD(func)	func(MMC, mmc, 1)
#else
	#define BOOT_TARGET_EMMC(func)
	#define BOOT_TARGET_MICROSD(func)
#endif

#if CONFIG_IS_ENABLED(CMD_NVME)
	#define BOOT_TARGET_NVME(func)	\
		func(NVME, nvme, 0)	\
		func(NVME, nvme, 1)
#else
	#define BOOT_TARGET_NVME(func)
#endif

#if CONFIG_IS_ENABLED(CMD_SCSI)
	#define BOOT_TARGET_SCSI(func)	func(SCSI, scsi, 0)
#else
	#define BOOT_TARGET_SCSI(func)
#endif
```